### PR TITLE
implement configurable dns refresh rate for strict dns clusters in envoy

### PR DESCRIPTION
**Pull request by GregHanson**
resolves https://github.com/istio/istio/issues/12181

### CODE REVIEW

**Comment by GregHanson** [comment]
cc @dwradcliffe @michaelkipper @duderino

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 1]
```diff
@@ -997,6 +997,11 @@ func buildDefaultCluster(env *model.Environment, name string, discoveryType apiv
 		cluster.DnsLookupFamily = apiv2.Cluster_V4_ONLY
 	}
 
+	if discoveryType == apiv2.Cluster_STRICT_DNS {
```
Comment (rshriram) [inline pilot/pkg/networking/core/v1alpha3/cluster.go]: or logical dns right? Not that we use logical DNS anywhere. if so, we should remove logical dns from the if block below

File: pilot/pkg/model/context.go [hunk 2]
```diff
@@ -420,6 +420,7 @@ func DefaultMeshConfig() meshconfig.MeshConfig {
 		DefaultVirtualServiceExportTo:     []string{"*"},
 		DefaultDestinationRuleExportTo:    []string{"*"},
 		OutboundTrafficPolicy:             &meshconfig.MeshConfig_OutboundTrafficPolicy{Mode: meshconfig.MeshConfig_OutboundTrafficPolicy_ALLOW_ANY},
+		DnsRefreshRate:                    types.DurationProto(5 * time.Second),
```
Comment (rshriram) [inline pilot/pkg/model/context.go]: can you add a comment here stating that this is the envoy default?

File: install/kubernetes/helm/istio/values.yaml [hunk 3]
```diff
@@ -168,6 +168,9 @@ global:
     # Configure the access log for sidecar to JSON or TEXT.
     accessLogEncoding: TEXT
 
+    # Configure the DNS refresh rate for Envoy cluster of type STRICT_DNS
+    dnsRefreshRate: 5s
```
Comment (rshriram) [inline install/kubernetes/helm/istio/values.yaml]: Also state here that this is the envoy default

**Comment by duderino** [comment]
@GregHanson nice one!

**Comment by duderino** [comment]
@GregHanson and please close the bug unless you are anticipating additional work on this one

**Pull request merged**